README.md

```diff
@@ -112,84 +125,141 @@ The content of your document is retrieved.
+This request supports the parameter below:
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`keyLengthInBits`|**[Number](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number)**| Optional  | Default `ENCRYPTION_PARAMETERS.keyLength` |
+
 
 ### encodeDocument
 
+This request performs encoding on a document.
+
 #### Parameters
 
-*   `document` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**&#x20;
+This request supports the parameter below:
+<!--Flag: pls check the necessity column, if it is optional or required.-->
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`document`|**[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**| Optional | Provide a document to encode it |
 
 ### decodeDocument
 
+This request performs decoding on an encoded document.
+
 #### Parameters
 
-*   `encoded` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**&#x20;
+This request supports the parameter below:
+<!--Flag: pls check the necessity column, if it is optional or required.-->
+
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|-------------|
+|`encoded`|**[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**| Optional | Provide an encoded document to decode it |
 
 ### IEncryptionResults
 
-### encryptString
+<!--Flag: pls clarify if we can add some description to explain IEncryptionResults -->
 
-Encrypts a given string with symmetric AES
+#### encryptString
 
-#### Parameters
+This request encrypts a given string with symmetric AES.
 
-*   `document` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** Input string to encrypt
-*   `key` **[string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)?**&#x20;
+##### Parameters
 
-Returns **[IEncryptionResults](#iencryptionresults)** cipherText cipher text in base64
+This request supports the parameters below:
+<!--Flag: pls check the necessity column, if it is optional or required.-->
 
-Returns **any** iv iv in base64
+| Parameter Name  | Data Type  | Necessity | Description |
+|-----------------|------------|-----------|------------|
+|`document`|**[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)**| Optional | Input a string to encrypt it |
+| `key`  | **[String](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)** | Optional | Input an encryption key |
 
-Returns **any** tag authenticated encryption tag in base64
+##### Results
 
-Returns **any** key encryption key in hexadecimal
+`IEncryptionResults` returns cipherText cipher text in base64:
```

Comment:
`IEncryptionResults` is what is being returned

Reply:
updated